gh-134201: Expand Base85 Documentation (#134288)
python/cpython: Doc/library/base64.rst

--------------

This module provides functions for encoding binary data to printable

> ASCII characters and decoding such encodings back to binary data,

Reviewer: ```suggestion
ASCII characters and decoding such encodings back to binary data.
```

To fix the ambiguous syntax below

> such as :ref:`those specified in RFC 4648 <base_64_rfc_4648>` and

Reviewer: ```suggestion
This includes the :ref:`encodings specified in <base64-rfc-4648>`
```

> :ref:`Base85 encodings <base_64_base_85>`.

Reviewer: ```suggestion
:rfc:`4648` (Base64, Base32 and Base16)
and the non-standard :ref:`Base85 encodings <base64-base-85>`.
```

Add a few clarifying details here and clean up the structure

There are two interfaces provided by this module.  The modern interface
supports encoding :term:`bytes-like objects <bytes-like object>` to ASCII
:class:`bytes`, and decoding :term:`bytes-like objects <bytes-like object>` or
strings containing ASCII to :class:`bytes`.  Both base-64 alphabets
defined in :rfc:`4648` (normal, and URL- and filesystem-safe) are supported.

> The :ref:`legacy interface <base_64_legacy>` does not support decoding from strings, but it does

Reviewer: ```suggestion
The :ref:`legacy interface <base-64-legacy>` does not support decoding from strings, but it does
```

This module provides functions for encoding binary data to printable
ASCII characters and decoding such encodings back to binary data.
This includes the :ref:`encodings specified in <base64-rfc-4648>`
:rfc:`4648` (Base64, Base32 and Base16)
and the non-standard :ref:`Base85 encodings <base64-base-85>`.

There are two interfaces provided by this module.  The modern interface
supports encoding :term:`bytes-like objects <bytes-like object>` to ASCII
:class:`bytes`, and decoding :term:`bytes-like objects <bytes-like object>` or
strings containing ASCII to :class:`bytes`.  Both base-64 alphabets
defined in :rfc:`4648` (normal, and URL- and filesystem-safe) are supported.

> The :ref:`legacy interface <base-64-legacy>` does not support decoding from strings, but it does

Reviewer: ```suggestion
The :ref:`legacy interface <base64-legacy>` does not support decoding from strings, but it does
```

   Any :term:`bytes-like objects <bytes-like object>` are now accepted by all
   encoding and decoding functions in this module.  Ascii85/Base85 support added.

> .. _base_64_rfc_4648:

Reviewer: ```suggestion
.. _base64-rfc-4648:
```

To match the rest of the module and general convention

RFC 4648 Encodings
------------------

The :rfc:`4648` encodings are suitable for encoding binary data so that it can be
safely sent by email, used as parts of URLs, or included as part of an HTTP

> POST request.  The encoding algorithm is not the same as the
> :program:`uuencode` program.

Reviewer: ```suggestion
POST request.
```

uuencode is long-obsolescent (removed from Python itself too) and not directly related to base64, so a reference here likely adds as much confusion as any it clarifies at this point

   incorrectly padded or if there are non-alphabet characters present in the
   input.

> .. _base_64_base_85:

Reviewer: ```suggestion
.. _base64-base-85:
```

Base85 Encodings
-----------------

> Base85 encoding is not formally specified. Base85 encoding is a de facto standard,
> and different systems perform the encoding differently.

Reviewer: ```suggestion
Base85 encoding is not formally specified but rather a de facto standard,
thus different systems perform the encoding differently.
```

Reduce duplication and improve phrasing/flow

> The Ascii85 and Base85 functions in this module are two implementations of

Reviewer 2: I'd do :func: / link to the functions rather than just saying name

Reviewer: (Implemented)

> the de facto standard. Which function you should use depends on how the other
> software you use implements the Base85 encoding.

Reviewer: ```suggestion
the de facto standard. You should call the function with the Base85
implementation used by the software you intend to work with.
```

   .. versionadded:: 3.13

> .. _base_64_legacy:

Reviewer: ```suggestion
.. _base64-legacy:
```
